### ISSUE TITLE
Initial boot hangs on ac-ispell install.

### ISSUE THREAD

**Issue by stephenprater** [syl20bnr/spacemacs#489]
`--debug-init` gives me this stack trace:

```elisp
Debugger entered--Lisp error: (wrong-type-argument arrayp ([cl-struct-package-desc ac-ispell (20140702 1553) "ispell completion source for auto-complete" ((auto-complete (1 4))) single "melpa" nil ((:url . "https://github.com/syohex/emacs-ac-ispell")) nil]))
  aref(([cl-struct-package-desc ac-ispell (20140702 1553) "ispell completion source for auto-complete" ((auto-complete (1 4))) single "melpa" nil ((:url . "https://github.com/syohex/emacs-ac-ispell")) nil]) 1)
  (if (version< emacs-version "24.4") (aref (cdr arch) 1) (let ((cl-x (cadr arch))) (progn (or (and (memq (aref cl-x 0) cl-struct-package-desc-tags)) (error "%s accessing a non-%s" (quote package-desc-reqs) (quote package-desc))) (aref cl-x 4))))
  (progn (if (version< emacs-version "24.4") (aref (cdr arch) 1) (let ((cl-x (cadr arch))) (progn (or (and (memq (aref cl-x 0) cl-struct-package-desc-tags)) (error "%s accessing a non-%s" (quote package-desc-reqs) (quote package-desc))) (aref cl-x 4)))))
  (if arch (progn (if (version< emacs-version "24.4") (aref (cdr arch) 1) (let ((cl-x (cadr arch))) (progn (or (and (memq ... cl-struct-package-desc-tags)) (error "%s accessing a non-%s" (quote package-desc-reqs) (quote package-desc))) (aref cl-x 4))))))
  (let* ((arch (assq pkg package-archive-contents)) (reqs (if arch (progn (if (version< emacs-version "24.4") (aref (cdr arch) 1) (let (...) (progn ... ...))))))) (progn (let ((--dolist-tail-- reqs) req) (while --dolist-tail-- (setq req (car --dolist-tail--)) (let* ((pkg2 (car req)) (reqs2 (configuration-layer//get-package-dependencies-from-archive pkg2))) (if reqs2 (progn (setq reqs ...)))) (setq --dolist-tail-- (cdr --$
  configuration-layer//get-package-dependencies-from-archive(ac-ispell)
  (let ((--dolist-tail-- (configuration-layer//get-package-dependencies-from-archive pkg)) dep) (while --dolist-tail-- (setq dep (car --dolist-tail--)) (configuration-layer//activate-package (car dep)) (setq --dolist-tail-- (cdr --dolist-tail--))))
  (progn (let ((--dolist-tail-- (configuration-layer//get-package-dependencies-from-archive pkg)) dep) (while --dolist-tail-- (setq dep (car --dolist-tail--)) (configuration-layer//activate-package (car dep)) (setq --dolist-tail-- (cdr --dolist-tail--)))))
  (if (not (assq pkg package-archive-contents)) (spacemacs/append-to-buffer (format "\nPackage %s is unavailable. Is the package name misspelled?\n" pkg)) (progn (let ((--dolist-tail-- (configuration-layer//get-package-dependencies-from-archive pkg)) dep) (while --dolist-tail-- (setq dep (car --dolist-tail--)) (configuration-layer//activate-package (car dep)) (setq --dolist-tail-- (cdr --dolist-tail--))))) (package-insta$
  (if (package-installed-p pkg) nil (if (not (assq pkg package-archive-contents)) (spacemacs/append-to-buffer (format "\nPackage %s is unavailable. Is the package name misspelled?\n" pkg)) (progn (let ((--dolist-tail-- (configuration-layer//get-package-dependencies-from-archive pkg)) dep) (while --dolist-tail-- (setq dep (car --dolist-tail--)) (configuration-layer//activate-package (car dep)) (setq --dolist-tail-- (cdr -$
  (while --dolist-tail-- (setq pkg (car --dolist-tail--)) (setq installed-count (1+ installed-count)) (spacemacs/replace-last-line-of-buffer (format "--> installing %s:%s... [%s/%s]" (ht-get configuration-layer-all-packages pkg) pkg installed-count not-installed-count) t) (if (package-installed-p pkg) nil (if (not (assq pkg package-archive-contents)) (spacemacs/append-to-buffer (format "\nPackage %s is unavailable. Is th$
  (let ((--dolist-tail-- not-installed) pkg) (while --dolist-tail-- (setq pkg (car --dolist-tail--)) (setq installed-count (1+ installed-count)) (spacemacs/replace-last-line-of-buffer (format "--> installing %s:%s... [%s/%s]" (ht-get configuration-layer-all-packages pkg) pkg installed-count not-installed-count) t) (if (package-installed-p pkg) nil (if (not (assq pkg package-archive-contents)) (spacemacs/append-to-buffer $
  (progn (let ((--dolist-tail-- not-installed) pkg) (while --dolist-tail-- (setq pkg (car --dolist-tail--)) (setq installed-count (1+ installed-count)) (spacemacs/replace-last-line-of-buffer (format "--> installing %s:%s... [%s/%s]" (ht-get configuration-layer-all-packages pkg) pkg installed-count not-installed-count) t) (if (package-installed-p pkg) nil (if (not (assq pkg package-archive-contents)) (spacemacs/append-to-$
  (progn (spacemacs/append-to-buffer (format "Found %s new package(s) to install...\n" not-installed-count)) (spacemacs/append-to-buffer "--> fetching new package repository indexes...\n") (redisplay) (package-refresh-contents) (setq installed-count 0) (progn (let ((--dolist-tail-- not-installed) pkg) (while --dolist-tail-- (setq pkg (car --dolist-tail--)) (setq installed-count (1+ installed-count)) (spacemacs/replace-la$
  (if not-installed (progn (spacemacs/append-to-buffer (format "Found %s new package(s) to install...\n" not-installed-count)) (spacemacs/append-to-buffer "--> fetching new package repository indexes...\n") (redisplay) (package-refresh-contents) (setq installed-count 0) (progn (let ((--dolist-tail-- not-installed) pkg) (while --dolist-tail-- (setq pkg (car --dolist-tail--)) (setq installed-count (1+ installed-count)) (sp$
  (let* ((not-installed (remove-if (quote package-installed-p) configuration-layer-all-packages-sorted)) (not-installed-count (length not-installed))) (if not-installed (progn (spacemacs/append-to-buffer (format "Found %s new package(s) to install...\n" not-installed-count)) (spacemacs/append-to-buffer "--> fetching new package repository indexes...\n") (redisplay) (package-refresh-contents) (setq installed-count 0) (pro$
  configuration-layer//install-packages()
  (let ((layers (reverse configuration-layer-layers))) (configuration-layer//set-layers-variables layers) (configuration-layer//load-layer-files layers (quote ("funcs.el" "config.el"))) (setq configuration-layer-excluded-packages (configuration-layer/get-excluded-packages layers)) (setq configuration-layer-all-packages (configuration-layer/get-packages layers)) (setq configuration-layer-all-pre-extensions (configuration-$
  configuration-layer/load-layers()
  (progn (load-file (concat user-emacs-directory "core/core-load-paths.el")) (require (quote core-spacemacs-mode)) (require (quote core-configuration-layer)) (dotspacemacs/load) (spacemacs/initialize) (dotspacemacs|call-func dotspacemacs/init) (configuration-layer/declare-all-layers) (configuration-layer/load-layers) (configuration-layer/delete-orphan-packages) (dotspacemacs|call-func dotspacemacs/config) (configuration-$
  (if (spacemacs/emacs-version-ok) (progn (load-file (concat user-emacs-directory "core/core-load-paths.el")) (require (quote core-spacemacs-mode)) (require (quote core-configuration-layer)) (dotspacemacs/load) (spacemacs/initialize) (dotspacemacs|call-func dotspacemacs/init) (configuration-layer/declare-all-layers) (configuration-layer/load-layers) (configuration-layer/delete-orphan-packages) (dotspacemacs|call-func dot$
  eval-buffer(#<buffer  *load*> nil "/Users/prater/.emacs.d/init.el" nil t)  ; Reading at buffer position 1364
  load-with-code-conversion("/Users/prater/.emacs.d/init.el" "/Users/prater/.emacs.d/init.el" t t)
  load("/Users/prater/.emacs.d/init" t t)
  #[0 "^H\205\262^@ \306=\203^Q^@\307^H\310Q\202;^@ \311=\204^^^@\307^H\312Q\202;^@\313\307\314\315#\203*^@\316\202;^@\313\307\314\317#\203:^@\320\nB^R\321\202;^@\316\322^S\323^A\322\211#\210^K\322=\203a^@\324\325\326\307^H\327Q!\"\323^A\322\211#\210^K\322=\203`^@^A^S\210^K\203\243^@\330^K!\331\232\203\243^@\332^K!\211\333P\334^A!\203}^@\211\202\210^@\334^B!\203\207^@^A\202\210^@\314\262^B^A\203\241^@\335^B^K\"\203\237^@$
  command-line()
  normal-top-level()
```

Which I am completely incapable of reading.

**Comment by syl20bnr** [comment]
What is your OS and Emacs version ?

A workaround is to copy the file `~/.emacs.d/core/templates/.spacemacs.template` to `~/.spacemacs` and add `ac-ispell` to `dotspacemacs-excluded-packages` variable.

**Comment by stephenprater** [comment]
OSX 10.10.1 - Emacs 24.3.50.1

**Comment by syl20bnr** [comment]
I think this is a spacemacs regression which breaks with `24.3`. I'll be able to investigate tonight.

**Comment by syl20bnr** [comment]
@stephenprater  Should be fixed in 0.51.0
